3.1.0 generates an error when using 3.0.0's configuration to generate HTML

### Contact Details

_No response_

### Version

3.1.0

### Plugin

smart-doc-maven-plugin

### Build tool version

maven 3.9.2

### Jdk version

1.8

### Describe the bug (Bug描述，社区已开启国际化推广，请用文心一言、讯飞星火等辅助翻译成英文，减少社区开发者的工作)

maven plugin config:
```
            <plugin>
                <groupId>com.ly.smart-doc</groupId>
                <artifactId>smart-doc-maven-plugin</artifactId>
                <version>3.1.0</version>
                <configuration>
                    <configFile>./src/main/resources/smart-doc.json</configFile>
                    <projectName>${project.description}</projectName>
                    <excludes>
                        <exclude>org.projectlombok:lombok</exclude>
                        <exclude>mysql:mysql-connector-java</exclude>
                        <exclude>com.fasterxml.jackson.datatype:jackson-datatype-hibernate5</exclude>
                        <exclude>com.google.guava:guava</exclude>
                        <exclude>org.redisson:redisson-spring-boot-starter</exclude>
                        <exclude>com.github.dozermapper:.*</exclude>
                        <exclude>org.apache.commons:.*</exclude>
                        <exclude>com.alibaba:.*</exclude>
                        <exclude>io.jsonwebtoken:.*</exclude>
                        <exclude>commons-net:.*</exclude>
                        <exclude>org.ehcache:*</exclude>
                        <exclude>com.github.ben-manes.caffeine:*</exclude>
                        <exclude>com.blazebit:*</exclude>
                        <exclude>com.h2database:*</exclude>
                        <exclude>com.github.wenhao:*</exclude>
                    </excludes>
                </configuration>
            </plugin>
```
smart-doc.json:
```json
{
  "allInOne": true,
  "isStrict": true,
  "serverUrl": "xxxx",
  "createDebugPage": true,
  "coverOld": true,
  "outPath": "doc",
  "packageFilters": "com.portal.controller.admin.v2.*",
  "appToken": "07a59d33595c41229a7d1cba7db7135b",
  "openUrl": "xxx",
  "debugEnvName":"测试环境",
  "replace": true,
  "author": "xxxxx",
  "requestHeaders": [
    {
      "name": "X-Auth-Token",
      "type": "string",
      "desc": "登录凭证",
      "value": "",
      "required": true,
      "since": "v1.0",
      "excludePathPatterns": "/**/a/v1/auth/login"
    }]
}
```
maven command:
```
mvn smart-doc:html -f pom.xml
```
get error:
```
[INFO] 
[INFO] --- smart-doc:3.1.0:markdown (default-cli) @ secureclient2 ---
[INFO] ------------------------------------------------------------------------
[INFO] Smart-doc Start preparing sources at: 2025-06-20 11:07:20
[INFO] Artifacts that the current project depends on: ["secureclient2"]
[INFO] Smart-doc has loaded the source code path: [{"path":"D:/wgw/workspace/aqkhd/backend/src/main/java"}]
[WARNING] The artifact mysql:mysql-connector-java:jar:8.0.33 has been relocated to com.mysql:mysql-connector-j:jar:8.0.33: MySQL Connector/J artifacts moved to reverse-DNS compliant Maven 2+ coordinates.
[INFO] Smart-doc Starting Create API Documentation at: 2025-06-20 11:07:24
[INFO] API documentation is output to => D:/wgw/workspace/aqkhd/backend/doc
[ERROR] 
java.lang.RuntimeException: Map's key can only use String or Enum for json,but you use java.lang.String
    at com.ly.doc.helper.JsonBuildHelper.buildMapJson (JsonBuildHelper.java:550)
    at com.ly.doc.helper.JsonBuildHelper.buildJson (JsonBuildHelper.java:241)
    at com.ly.doc.helper.JsonBuildHelper.buildReturnJson (JsonBuildHelper.java:120)
    at com.ly.doc.template.IRestDocTemplate.buildExceptionStatus (IRestDocTemplate.java:679)
    at com.ly.doc.template.IRestDocTemplate.processApiData (IRestDocTemplate.java:185)
    at com.ly.doc.template.SpringBootDocBuildTemplate.renderApi (SpringBootDocBuildTemplate.java:63)
    at com.ly.doc.template.IDocBuildTemplate.getApiData (IDocBuildTemplate.java:56)
    at com.ly.doc.builder.ApiDocBuilder.buildApiDoc (ApiDocBuilder.java:71)
    at com.ly.doc.plugin.mojo.MarkDownMojo.executeMojo (MarkDownMojo.java:44)
    at com.ly.doc.plugin.mojo.BaseDocsGeneratorMojo.execute (BaseDocsGeneratorMojo.java:163)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:910)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
```


### Expected Behavior (您期望的结果，社区已开启国际化推广，请用文心一言、讯飞星火等辅助翻译成英文，减少社区开发者的工作)

generate doc without error

### Current Behavior (当前结果，社区已开启国际化推广，请用文心一言、讯飞星火等辅助翻译成英文，减少社区开发者的工作)

error

### Steps to Reproduce (Bug产生步骤，请尽量提供用例代码。社区已开启国际化推广，请用文心一言、讯飞星火等辅助翻译成英文，减少社区开发者的工作)

I used smart-doc 3.0.0 version to generate normally, replaced with 3.1.0 and it went wrong, please help me

### Possible Solution (Bug解决建议，社区已开启国际化推广，请用文心一言、讯飞星火等辅助翻译成英文，减少社区开发者的工作)

_No response_

### Context (Bug影响描述，社区已开启国际化推广，请用文心一言、讯飞星火等辅助翻译成英文，减少社区开发者的工作)

_No response_

### Validations

- [x] Check if you're on the [latest smart-doc version](https://github.com/TongchengOpenSource/smart-doc/releases/latest). (请检查是否为最新版本)
- [x] Read the [docs](https://smart-doc-group.github.io/). (请先阅读官方文档)
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate. (检查是否存在报告相同错误的问题，以避免产生重复)

--- a/src/main/java/com/ly/doc/helper/JsonBuildHelper.java
+++ b/src/main/java/com/ly/doc/helper/JsonBuildHelper.java
@@ -545,10 +545,14 @@ public class JsonBuildHelper extends BaseHelper {
 		}
 		JavaClass mapKeyClass = builder.getJavaProjectBuilder().getClassByName(getKeyValType[0]);
 		boolean mapKeyIsEnum = mapKeyClass.isEnum();
-		if ((!JavaTypeConstants.JAVA_STRING_FULLY.equals(getKeyValType[0]) || !mapKeyIsEnum
-				|| mapKeyClass.getEnumConstants().isEmpty()) && builder.getApiConfig().isStrict()) {
-			throw new RuntimeException(
-					"Map's key can only use String or Enum for json,but you use " + getKeyValType[0]);
+		String keyType = getKeyValType[0];
+
+		if (builder.getApiConfig().isStrict()) {
+			boolean isStringKey = JavaTypeConstants.JAVA_STRING_FULLY.equals(keyType);
+			boolean isValidEnumKey = mapKeyIsEnum && !mapKeyClass.getEnumConstants().isEmpty();
+			if (!isStringKey && !isValidEnumKey) {
+				throw new RuntimeException("Map's key can only use String or Enum for JSON, but you used: " + keyType);
+			}
 		}
 		String gicName = getKeyValType[1];
 		// when map key is enum

